cluster-autoscaler/cloudprovider/konvoy/konvoy_test.go

```diff
@@ -0,0 +1,128 @@
+package konvoy
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	kubernetesfake "k8s.io/client-go/kubernetes/fake"
+	utilpointer "k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	konvoyv1beta1 "github.com/mesosphere/konvoy/clientapis/pkg/apis/konvoy/v1beta1"
+	konvoyclusterv1beta1 "github.com/mesosphere/yakcl/clientapis/pkg/apis/kommander/v1beta1"
+	yakclv1beta1 "github.com/mesosphere/yakcl/clientapis/pkg/apis/kommander/v1beta1"
+	kube_util "k8s.io/autoscaler/cluster-autoscaler/utils/kubernetes"
+)
+
+var testKonvoyCluster = &yakclv1beta1.KonvoyCluster{
+	ObjectMeta: metav1.ObjectMeta{
+		Name:      "test-cluster",
+		Namespace: "kommander",
+	},
+	Spec: yakclv1beta1.KonvoyClusterSpec{
+		ProvisioningPaused: false,
+		ProvisionerConfiguration: konvoyv1beta1.ClusterProvisionerSpec{
+			NodePools: []konvoyv1beta1.MachinePool{
+				{
+					Name: "test-autoscaling-pool",
+					AutoscalingOptions: &konvoyv1beta1.AutoscalingOptions{
+						MinSize: utilpointer.Int32Ptr(1),
+						MaxSize: utilpointer.Int32Ptr(10),
+					},
+				},
+			},
+		},
+	},
+}
+
+func TestKonvoyNodeGroupForNode(t *testing.T) {
+
+	var tests = []struct {
+		description   string
+		expectedError bool
+		node          *corev1.Node
+		nodeGroup     NodeGroup
+	}{
+		{
+			description:   "should return nil node group due to a master node",
+			expectedError: false,
+			node: &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-node",
+					Labels: map[string]string{
+						KubernetesMasterNodeLabel: "master",
+					},
+				},
+				Spec: corev1.NodeSpec{
+					ProviderID: "test-provider-id",
+				},
+			},
+		},
+		{
+			description: "should return the group name",
+			nodeGroup: NodeGroup{
+				Name: "test-node-group",
+			},
+			expectedError: false,
+			node: &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-node",
+					Labels: map[string]string{
+						nodeGroupLabel: "test-node-group",
+					},
+				},
+			},
+		},
+		{
+			description:   "should return nil node group due to an unknown node",
```

Comment:
this is due to missing `nodeGroupLabel` that is required to resolve node group.

Reply:
i will change the description